airflow/providers/amazon/aws/operators/redshift_cluster.py

```diff
@@ -392,20 +393,31 @@ def __init__(
         *,
         cluster_identifier: str,
         aws_conn_id: str = "aws_default",
+        attempts: int = 1,
+        attempt_interval: int = 30,
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.cluster_identifier = cluster_identifier
         self.aws_conn_id = aws_conn_id
+        self.attempts = attempts
+        self.attempt_interval = attempt_interval
 
     def execute(self, context: Context):
         redshift_hook = RedshiftHook(aws_conn_id=self.aws_conn_id)
-        cluster_state = redshift_hook.cluster_status(cluster_identifier=self.cluster_identifier)
-        if cluster_state == "paused":
-            self.log.info("Starting Redshift cluster %s", self.cluster_identifier)
-            redshift_hook.get_conn().resume_cluster(ClusterIdentifier=self.cluster_identifier)
-        else:
-            raise Exception(f"Unable to resume cluster - cluster state is {cluster_state}")
+
+        while self.attempts >= 1:
+            try:
+                redshift_hook.get_conn().resume_cluster(ClusterIdentifier=self.cluster_identifier)
+                return
+            except redshift_hook.get_conn().exceptions.InvalidClusterStateFault as error:
+                self.attempts = self.attempts - 1
+
+                if self.attempts > 0:
+                    self.log.error("Unable to resume cluster. %d attempts remaining.", self.attempts)
+                    time.sleep(self.attempt_interval)
+                else:
+                    raise error
```

Comment:
This is essentially the behavior of `retries` and `retry_delay` of BaseOperator

Why do we need another layer of "internal retries"?
I can argue that if we have such need then it possibly goes beyond this single operator and may apply to others.
If we must have it lets not expose the parameters to the users. I'm not sure if we want each indvidual operator to set it's own user facing logic for retry.

I think this is something should be discussed in more depth. I'd like to hear others on this one.

Reply:
The reason the retries are necessary is because it's not clear why the cluster initially fails to resume (or pause). Despite the state of the cluster being in the correct state to perform the operation, the first few attempts fail (depending on the `attempt_interval`). 
You have a good point about the behaviour of `retries` and `retry_delay`. Should I just use those, and remove `attempts` and `attempt_interval`?
> If we must have it lets not expose the parameters to the users. I'm not sure if we want each indvidual operator to set it's own user facing logic for retry.

I'm not sure what you mean by this.

Reply:
IMHO the top level `retries` and `retry_delay` are for users to generally dictate what an operator should do when it unexpectedly fails. The code Syed is adding here is to actually account for a known poor behaviour on the AWS redshift API/SDK where it unreliably reports a good state but still rejects mutations to the cluster for a short time.

Though I do agree with @eladkal's second point, we don't need to expose the parameters to the users. @syedahsn we can drop the arguments you added to the method stub and just define them locally here with reasonable defaults.

Reply:
> The code Syed is adding here is to actually account for a known poor behaviour on the AWS redshift API/SDK where it unreliably reports a good state but still rejects mutations to the cluster for a short time.

I agree. I see this in same way as we handle random http failures for example:
https://github.com/apache/airflow/blob/3d5f34cb0f294d21dd1ba244af0fa06873377f11/airflow/providers/google/cloud/hooks/dataprep.py#L77-L78

Reply:
> > The code Syed is adding here is to actually account for a known poor behaviour on the AWS redshift API/SDK where it unreliably reports a good state but still rejects mutations to the cluster for a short time.
> 
> I agree. I see this in same way as we handle random http failures for example:
> 
> https://github.com/apache/airflow/blob/3d5f34cb0f294d21dd1ba244af0fa06873377f11/airflow/providers/google/cloud/hooks/dataprep.py#L77-L78

I really like the `@retry` decorator approach too, we should use that more often within airflow codebase.